coriolis/tests/worker/rpc/test_server.py

```diff
@@ -192,3 +265,297 @@ def test_get_endpoint_instance(self, mock_get_provider, mock_get_secret, mock_va
+            mock.sentinel.pool_environment)
+
+        mock_get_provider.assert_called_once_with(
+            mock.sentinel.destination_platform_name, constants.PROVIDER_TYPE_DESTINATION_MINION_POOL, None)
+        mock_validate.assert_called_once_with(
+            mock.sentinel.pool_environment, mock_get_provider.return_value.get_minion_pool_environment_schema.return_value)
+
+        self.assertEquals(result, (True, None))
+
+        # handle SchemaValidationException
+        mock_validate.side_effect = exception.SchemaValidationException('test')
+        result = self.server.validate_endpoint_destination_minion_pool_options(
+            mock.sentinel.context, mock.sentinel.destination_platform_name,
+            mock.sentinel.pool_environment)
+
+        self.assertEquals(result, (False, 'test'))
+
+    @mock.patch.object(schemas, 'validate_value')
+    @mock.patch.object(providers_factory, 'get_provider')
+    @mock.patch.object(utils, 'get_secret_connection_info')
+    def test_validate_endpoint_connection(self, mock_get_secret, mock_get_provider, mock_validate):
+        def call_validate_endpoint_connection():
+            return self.server.validate_endpoint_connection(
+                mock.sentinel.context, mock.sentinel.platform_name,
+                mock.sentinel.connection_info)
+
+        result = call_validate_endpoint_connection()
+
+        mock_get_provider.assert_called_once_with(
+            mock.sentinel.platform_name, constants.PROVIDER_TYPE_ENDPOINT, None)
+        mock_validate.assert_called_once_with(
+            mock_get_secret.return_value, mock_get_provider.return_value.get_connection_info_schema.return_value)
+        mock_get_provider.return_value.validate_connection.assert_called_once_with(
+            mock.sentinel.context, mock_get_secret.return_value)
+
+        self.assertEquals(result, (True, None))
+
+        # handle SchemaValidationException
+        mock_validate.side_effect = exception.SchemaValidationException()
+        result = call_validate_endpoint_connection()
+        self.assertEquals(result, (False, (
+            "Schema validation for the provided connection parameters has "
+            "failed. Please ensure that you have included all the "
+            "necessary connection parameters and they are all properly "
+            "formatted for the '%s' Coriolis plugin in use." % (
+                mock.sentinel.platform_name))))
+
+        # handle ConnectionValidationException
+        mock_validate.side_effect = exception.ConnectionValidationException()
+        result = call_validate_endpoint_connection()
+        self.assertEquals(result, (False, 'An unknown exception occurred.'))
+
+        # handle BaseException
+        mock_validate.side_effect = BaseException('test')
+        result = call_validate_endpoint_connection()
+        self.assertEquals(
+            result, (False, 'An unexpected connection validation exception ocurred: test'))
```

Comment:
Ditto. Please check and remove all other occurrences of exception message equal assertions.